### PR TITLE
Add test for net-list options

### PR DESCRIPTION
**Pull request by Grace-chhu**
- New options: uuid, name, table
- Other options: persistent, transient, autostart, no-autostart

Signed-off-by: chhu <chhu@redhat.com>

### CODE REVIEW

**Comment by Grace-chhu** [comment]
# avocado run --vt-type libvirt virsh.net_list
JOB ID     : 3e72d4e71e9427dbb7e514b45d9066a9bdbcc1e3
JOB LOG    : /root/avocado/job-results/job-2016-11-29T14.39-3e72d4e/job.log
TESTS      : 35
 (01/35) type_specific.io-github-autotest-libvirt.virsh.net_list.no_option: PASS (0.83 s)
 (02/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.non_acl.inactive_option.inactive_status: PASS (0.63 s)
 (03/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.non_acl.all_option: PASS (0.80 s)
 (04/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.non_acl.persistent_option.persistent_status: PASS (0.82 s)
 (05/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.non_acl.transient_option.transient_status: PASS (0.73 s)
 (06/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.non_acl.autostart_option.autostart_status: PASS (0.81 s)
 (07/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.non_acl.noautostart_option.noautostart_status: PASS (0.75 s)
 (08/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.non_acl.uuid_option: PASS (0.83 s)
 (09/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.non_acl.name_option: PASS (0.81 s)
 (10/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.non_acl.table_option: PASS (0.82 s)
 (11/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.non_acl.combined_option.all_persistent: PASS (0.82 s)
 (12/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.non_acl.combined_option.all_noautostart: PASS (0.74 s)
 (13/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.non_acl.combined_option.persistent_inactive: PASS (0.63 s)
 (14/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.acl_test.inactive_option.inactive_status: \PolicyKit daemon disconnected from the bus.
We are no longer a registered authentication agent.
PASS (1.92 s)
 (15/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.acl_test.all_option: /PolicyKit daemon disconnected from the bus.
We are no longer a registered authentication agent.
PASS (2.09 s)
 (16/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.acl_test.persistent_option.persistent_status: \PolicyKit daemon disconnected from the bus.
We are no longer a registered authentication agent.
PASS (2.13 s)
 (17/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.acl_test.transient_option.transient_status: /PolicyKit daemon disconnected from the bus.
We are no longer a registered authentication agent.
PASS (2.02 s)
 (18/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.acl_test.autostart_option.autostart_status: \PolicyKit daemon disconnected from the bus.
We are no longer a registered authentication agent.
PASS (2.11 s)
 (19/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.acl_test.noautostart_option.noautostart_status: /PolicyKit daemon disconnected from the bus.
We are no longer a registered authentication agent.
PASS (2.13 s)
 (20/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.acl_test.uuid_option: \PolicyKit daemon disconnected from the bus.
We are no longer a registered authentication agent.
PolicyKit daemon reconnected to bus.
Attempting to re-register as an authentication agent.
PASS (2.09 s)
 (21/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.acl_test.name_option: /PolicyKit daemon disconnected from the bus.
We are no longer a registered authentication agent.
PolicyKit daemon reconnected to bus.
Attempting to re-register as an authentication agent.
PASS (2.05 s)
 (22/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.acl_test.table_option: \PolicyKit daemon disconnected from the bus.
We are no longer a registered authentication agent.
PASS (2.15 s)
 (23/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.acl_test.combined_option.all_persistent: -PolicyKit daemon disconnected from the bus.
We are no longer a registered authentication agent.
PolicyKit daemon reconnected to bus.
Attempting to re-register as an authentication agent.
PASS (2.11 s)
 (24/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.acl_test.combined_option.all_noautostart: |PolicyKit daemon disconnected from the bus.
We are no longer a registered authentication agent.
PASS (1.97 s)
 (25/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.acl_test.combined_option.persistent_inactive: -PolicyKit daemon disconnected from the bus.
We are no longer a registered authentication agent.
PASS (1.87 s)
 (26/35) type_specific.io-github-autotest-libvirt.virsh.net_list.error_test.inactive_option.active_status: PASS (0.84 s)
 (27/35) type_specific.io-github-autotest-libvirt.virsh.net_list.error_test.persistent_option.transient_status: PASS (0.75 s)
 (28/35) type_specific.io-github-autotest-libvirt.virsh.net_list.error_test.transient_option.persistent_status: PASS (0.83 s)
 (29/35) type_specific.io-github-autotest-libvirt.virsh.net_list.error_test.autostart_option.noautostart_status: PASS (0.76 s)
 (30/35) type_specific.io-github-autotest-libvirt.virsh.net_list.error_test.noautostart_option.autostart_status: PASS (0.81 s)
 (31/35) type_specific.io-github-autotest-libvirt.virsh.net_list.error_test.combined_option.uuid_name: PASS (0.83 s)
 (32/35) type_specific.io-github-autotest-libvirt.virsh.net_list.error_test.combined_option.uuid_table: PASS (0.80 s)
 (33/35) type_specific.io-github-autotest-libvirt.virsh.net_list.error_test.combined_option.name_table: PASS (0.82 s)
 (34/35) type_specific.io-github-autotest-libvirt.virsh.net_list.error_test.extra_args_option: PASS (0.81 s)
 (35/35) type_specific.io-github-autotest-libvirt.virsh.net_list.error_test.extra_option: PASS (0.82 s)
RESULTS    : PASS 35 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0
JOB HTML   : /root/avocado/job-results/job-2016-11-29T14.39-3e72d4e/html/results.html
TESTS TIME : 42.71 s

File: libvirt/tests/src/virsh_cmd/network/virsh_net_list.py [hunk 1]
```diff
@@ -45,104 +40,89 @@ def run(test, params, env):
 
     if not libvirt_version.version_compare(1, 1, 1):
         if params.get('setup_libvirt_polkit') == 'yes':
-            raise error.TestNAError("API acl test not supported in current"
-                                    " libvirt version.")
+            raise exceptions.NotATestError("API acl test not supported"
```
Comment (waynesun09) [inline libvirt/tests/src/virsh_cmd/network/virsh_net_list.py]: raise TestSkipError here

File: libvirt/tests/cfg/virsh_cmd/network/virsh_net_list.cfg [hunk 2]
```diff
@@ -38,7 +74,43 @@
...
+                    variants:
+                        - uuid_name:
+                            net_list_extra = "--uuid --name"
+                        - uuid_table:
+                            net_list_extra = "--uuid --table"
+                        - name_table:
+                            net_list_extra = "--name --table"
+                - extra_args_option:
+                    net_list_option = ""
+                    net_list_extra = "--xyz"
```
Comment (waynesun09) [inline libvirt/tests/cfg/virsh_cmd/network/virsh_net_list.cfg]: No need for this 'extra' in testing, just use net_list_option will be enough since this is negative testing

File: libvirt/tests/src/virsh_cmd/network/virsh_net_list.py [hunk 3]
```diff
@@ -45,104 +40,89 @@ def run(test, params, env):
...
+        if params.get('setup_libvirt_polkit') == 'yes':
+            virsh_dargs['unprivileged_user'] = unprivileged_user
+            virsh_dargs['uri'] = uri
+        result = virsh.net_list(option, extra, **virsh_dargs)
+        status = result.exit_status
+        output = result.stdout.strip()
+        err = result.stderr.strip()
+
+        # Check test results
+        if status_error == "yes":
```
Comment (waynesun09) [inline libvirt/tests/src/virsh_cmd/network/virsh_net_list.py]: use check_exit_status at following check

File: libvirt/tests/cfg/virsh_cmd/network/virsh_net_list.cfg [hunk 4]
```diff
@@ -7,38 +7,113 @@
...
+    set_status = "active"
+    set_persistent = "persistent"
+    set_autostart = "autostart"
     requires_root = "yes"
+    error_msg = ""
     variants:
+        - no_option:
         - normal_test:
-            status_error = "no"
+            negative_test = "no"
```
Comment (waynesun09) [inline libvirt/tests/cfg/virsh_cmd/network/virsh_net_list.cfg]: no need to change status_error with new name negative_test, status_error have been the flag to identify the expect cmd result in most cases

File: libvirt/tests/src/virsh_cmd/network/virsh_net_list.py [hunk 5]
```diff
@@ -45,104 +42,79 @@ def run(test, params, env):
 
     if not libvirt_version.version_compare(1, 1, 1):
         if params.get('setup_libvirt_polkit') == 'yes':
-            raise error.TestNAError("API acl test not supported in current"
-                                    " libvirt version.")
+            raise exceptions.TestskipError("API acl test not supported"
```
Comment (balamuruhans) [inline libvirt/tests/src/virsh_cmd/network/virsh_net_list.py]: typo %s/exceptions.TestskipError/exceptions.TestSkipError

File: libvirt/tests/src/virsh_cmd/network/virsh_net_list.py [hunk 6]
```diff
@@ -45,104 +42,79 @@ def run(test, params, env):
...
-    virsh_dargs = {'ignore_status': True}
-    if params.get('setup_libvirt_polkit') == 'yes':
-        virsh_dargs['unprivileged_user'] = unprivileged_user
-        virsh_dargs['uri'] = uri
-    result = virsh.net_list(option, extra, **virsh_dargs)
-    status = result.exit_status
-    output = result.stdout.strip()
-
-    # Recover network
+        raise exceptions.NotATestError("Duplicated network name: '%s'" %
```
Comment (balamuruhans) [inline libvirt/tests/src/virsh_cmd/network/virsh_net_list.py]: TestSkipError can be used here.

**Comment by Grace-chhu** [comment]
# avocado run --vt-type libvirt virsh.net_list
JOB ID     : 1ef658189ee42685002569a2fcc4e7492bdeafa5
JOB LOG    : /root/avocado/job-results/job-2016-12-08T13.19-1ef6581/job.log
TESTS      : 35
 (01/35) type_specific.io-github-autotest-libvirt.virsh.net_list.no_opt: PASS (0.83 s)
 (02/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.non_acl.inactive_opt: PASS (0.61 s)
 (03/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.non_acl.all_opt: PASS (0.78 s)
 (04/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.non_acl.persistent_opt: PASS (0.78 s)
 (05/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.non_acl.transient_opt: PASS (0.71 s)
 (06/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.non_acl.autostart_opt: PASS (0.79 s)
 (07/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.non_acl.noautostart_opt: PASS (0.74 s)
 (08/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.non_acl.uuid_opt: PASS (0.80 s)
 (09/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.non_acl.name_opt: PASS (0.77 s)
 (10/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.non_acl.table_opt: PASS (0.78 s)
 (11/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.non_acl.combined.all_persistent: PASS (0.77 s)
 (12/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.non_acl.combined.all_noautostart: PASS (0.73 s)
 (13/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.non_acl.combined.persistent_inactive: PASS (0.60 s)
 (14/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.acl_test.inactive_opt: \PolicyKit daemon disconnected from the bus.
We are no longer a registered authentication agent.
PolicyKit daemon reconnected to bus.
Attempting to re-register as an authentication agent.
PASS (1.86 s)
 (15/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.acl_test.all_opt: /PolicyKit daemon disconnected from the bus.
We are no longer a registered authentication agent.
PolicyKit daemon reconnected to bus.
Attempting to re-register as an authentication agent.
PASS (2.11 s)
 (16/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.acl_test.persistent_opt: \PolicyKit daemon disconnected from the bus.
We are no longer a registered authentication agent.
PolicyKit daemon reconnected to bus.
Attempting to re-register as an authentication agent.
We are now a registered authentication agent.
PASS (2.12 s)
 (17/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.acl_test.transient_opt: /PolicyKit daemon disconnected from the bus.
We are no longer a registered authentication agent.
PASS (1.99 s)
 (18/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.acl_test.autostart_opt: \PolicyKit daemon disconnected from the bus.
We are no longer a registered authentication agent.
PolicyKit daemon reconnected to bus.
Attempting to re-register as an authentication agent.
PASS (2.13 s)
 (19/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.acl_test.noautostart_opt: /PolicyKit daemon disconnected from the bus.
We are no longer a registered authentication agent.
PolicyKit daemon reconnected to bus.
Attempting to re-register as an authentication agent.
PASS (1.98 s)
 (20/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.acl_test.uuid_opt: \PolicyKit daemon disconnected from the bus.
We are no longer a registered authentication agent.
PASS (2.08 s)
 (21/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.acl_test.name_opt: /PolicyKit daemon disconnected from the bus.
We are no longer a registered authentication agent.
PASS (2.07 s)
 (22/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.acl_test.table_opt: \PolicyKit daemon disconnected from the bus.
We are no longer a registered authentication agent.
PolicyKit daemon reconnected to bus.
Attempting to re-register as an authentication agent.
PASS (2.10 s)
 (23/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.acl_test.combined.all_persistent: /PolicyKit daemon disconnected from the bus.
We are no longer a registered authentication agent.
PASS (2.63 s)
 (24/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.acl_test.combined.all_noautostart: |PolicyKit daemon disconnected from the bus.
We are no longer a registered authentication agent.
PASS (1.98 s)
 (25/35) type_specific.io-github-autotest-libvirt.virsh.net_list.normal_test.acl_test.combined.persistent_inactive: -PolicyKit daemon disconnected from the bus.
We are no longer a registered authentication agent.
PASS (1.90 s)
 (26/35) type_specific.io-github-autotest-libvirt.virsh.net_list.error_test.inactive_opt: PASS (0.84 s)
 (27/35) type_specific.io-github-autotest-libvirt.virsh.net_list.error_test.persistent_opt: PASS (0.73 s)
 (28/35) type_specific.io-github-autotest-libvirt.virsh.net_list.error_test.transient_opt: PASS (0.82 s)
 (29/35) type_specific.io-github-autotest-libvirt.virsh.net_list.error_test.autostart_opt: PASS (0.75 s)
 (30/35) type_specific.io-github-autotest-libvirt.virsh.net_list.error_test.noautostart_opt: PASS (0.80 s)
 (31/35) type_specific.io-github-autotest-libvirt.virsh.net_list.error_test.combined.uuid_name: PASS (0.79 s)
 (32/35) type_specific.io-github-autotest-libvirt.virsh.net_list.error_test.combined.uuid_table: PASS (0.79 s)
 (33/35) type_specific.io-github-autotest-libvirt.virsh.net_list.error_test.combined.name_table: PASS (0.79 s)
 (34/35) type_specific.io-github-autotest-libvirt.virsh.net_list.error_test.extra_args_opt: PASS (1.00 s)
 (35/35) type_specific.io-github-autotest-libvirt.virsh.net_list.error_test.extra_opt: PASS (0.78 s)
RESULTS    : PASS 35 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0
JOB HTML   : /root/avocado/job-results/job-2016-12-08T13.19-1ef6581/html/results.html
TESTS TIME : 42.74 s
